mysql-test/suite/mysql_migration/mysql_migration.sql

```diff
@@ -0,0 +1,108 @@
+-- This is needed for mariadb-upgrade to work
+CREATE DATABASE test;
+
+
+-- Create tables partitioned with "native" InnoDb partition handler
+-- Later will test reading them with MariaDB
+CREATE DATABASE idb_partition_db1;
+USE idb_partition_db1;
+
+CREATE TABLE idb_partition_t1 (
+    id INT,
+    year INT
+);
+
+ALTER TABLE idb_partition_t1
+    PARTITION BY HASH(id)
+    PARTITIONS 4;
+
+INSERT INTO idb_partition_t1(id, year)
+VALUES
+    (0,1997),
+    (1,1998),
+    (2,1999),
+    (3,2000);
+
+
+-- Create a table compressed with MySQL punch-hole compression
+-- Later will test reading them with MariaDB
+CREATE DATABASE idb_page_comp_db1;
+USE idb_page_comp_db1;
+
+CREATE TABLE idb_page_comp_t1
+    (id INT,
+    year INT)
+    COMPRESSION="zlib";
+
+-- Insert large data to make sure the compression will take place
+DELIMITER $$
+CREATE PROCEDURE populatePageCompTable()
```

Comment:
It could be good to test that procedure definitions are the same after upgrading. I do not see the name of this procedure anywhere except this statement and the subsequent `call` statement.

Reply:
> It could be good to test that procedure definitions are the same after upgrading.

In order to test that thoroughly, all different kinds of procedures and metadata (e.g. `SQL SECURITY DEFINER` vs. `… INVOKER`, `DETERMINISTIC`, etc.) would have to be created and verified.

That does seem like a good idea for coverage, but perhaps a bit outside the scope of this PR, which is focused on documenting the already-known cases of incompatibilities in MySQL 5.7 → MariaDB data directory migration.

Reply:
Sure, there could be better frameworks for testing the compatibility of stored procedures. Perhaps something like the Random Query Generator could be employed to cover the entire MySQL 5.7 grammar for stored procedures, and perhaps we should also test that the execution is compatible. I agree that something like it is out of the scope of this PR.

Still, it might be better to not waste the opportunity to ensure that `SHOW PROCEDURE`  produces the expected output for these few special cases.